pkg/catalog/walk_entry_iterator.go

```diff
@@ -0,0 +1,156 @@
+package catalog
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/treeverse/lakefs/pkg/ingest/store"
+	"go.uber.org/atomic"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type walkEntryIterator struct {
+	entries chan EntryWithMarker
+	walker  *store.WalkerWrapper
+
+	done   chan bool
+	closed *atomic.Bool
+	err    *atomic.Error
+
+	curr EntryWithMarker
+}
+
+// Mark stands for pagination information when listing objects from the blockstore.
+// It is used for server-client communication on the status of range ingestion.
+type Mark store.Mark
+
+type EntryWithMarker struct {
+	EntryRecord
+	Mark
+}
+
+var (
+	// ErrItClosed is used to determine the reason for the end of the walk
+	ErrItClosed = errors.New("iterator closed")
+
+	errSeekGENotSupported = errors.New("SeekGE not supported for walk entry iterator")
+)
+
+// buffer size of the buffer between reading entries from the blockstore Walk and passing it on
+const bufferSize = 100
```

Comment:
code doc starts with the name
```suggestion
// bufferSize - buffer size of the buffer between reading entries from the blockstore Walk and passing it on
const bufferSize = 100
```

Reply:
Done.